Use datasource fetched from remote server if available to check if Data Registry Report

gherceg opened the pull request:
## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->

## Technical Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->
https://dimagi-dev.atlassian.net/browse/SUPPORT-15186

Introduced in https://github.com/dimagi/commcare-hq/pull/31682

The check for a report config being associated with a `DataRegistry` assumes the report is hosted on the same server, not a remote server. This is a quick fix to prevent remote links from failing to be pulled due to this check. If there is a use case that calls for it, the better solution is to support data registry reports in remote links as well, but it seems that moving the needle from no remote report links working to non data registry remote report links working is worthwhile.

**Update**: after Ethan's feedback, we had the datasource available from the remote server, and could use that to perform the data registry check.

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

gherceg committed:
only check for data registry if not remote link

esoergel commented on corehq/apps/linked_domain/ucr.py:
```diff
@@ -63,7 +63,7 @@ def create_linked_ucr(domain_link, report_config_id):
                 'The data source referenced by this report could not be found.'
             ))
-    if is_data_registry_report(report_config):
+    if not domain_link.is_remote and is_data_registry_report(report_config):
```
What will happen if someone _does_ attempt to remote link a data registry report?  With this change, it'll attempt to proceed, whereas I think it'd be more correct to raise the `RegistryNotAccessible` error that you see a few lines below (maybe with a different message, about it being remote).

As I understand the problem, it's that the `is_data_registry_report` check fails because it includes the line `report_config.config.meta.build.registry_slug`, and `report_config.config` attempts to fetch the datasource config from the DB.  It of course isn't there, because it's a remote server.  However, we do have that config already, in the local variable `datasource`, so we could rework things to instead check `datasource.meta.build.registry_slug` in this workflow to preserve the check.

gherceg replied:
Yeah this is a great point, and sounds pretty straightforward to fix. I'll do that shortly.

gherceg replied:
60837290003ca199ca15dce8aa13e1ebdebc9d6c

gherceg replied:
Whoops forgot to think about other usages of `is_data_registry_report`: d234d429c1dc4b42351d11581c712e8defa9e8d7

esoergel reviewed (commented):
I believe this will also fix this USH issue: https://dimagi-dev.atlassian.net/browse/USH-2546
(Thanks @orangejenny for calling this PR to my attention)

gherceg committed:
use datasource var in data registry check

the datasource var is populated whether the link is local or remote
and can be used to verify if this is a data registry report

gherceg committed:
datasource as optional param in is_data_registry_report

esoergel reviewed (approved):
excellent, thanks!
